enhancements/monitoring/user-workload-monitoring.md

```diff
@@ -173,27 +184,96 @@ Usage of the Thanos querier component inside cluster monitoring has many advanta
 
 See [5] and [6] for more details.
 
-### Tenancy
+#### Alerts/Recording Rules Aggregation
 
-![](user-monitoring-request.png)
+Thanos Querier only solves aggregating the query path. For aggregating recording rules as well as alerting rules
+another component [Thanos Ruler](https://thanos.io/components/rule.md/) is deployed. This component is responsible for hosting user defined recording rules
+as well as alerting rules.
+
+Thanos Ruler is deployed via the user workload monitoring prometheus operator via a new CRD called ThanosRule.
+The same _deny list_ setting is applied to user workload monitoring prometheus operator such that the user cannot
+deploy its own Thanos Ruler instances in user namespaces.
+
+Thanos Ruler queries Thanos Querier to resolve recording and alerting rules.
+This does have implications on the availability of user defined alerts.
+The following list enumerates critical components for alerts to work for this use case:
+
+1. Thanos Ruler
+2. Thanos Querier
+3. User workload Prometheus
+4. Cluster monitoring Prometheus
+
+If there is no need to aggregate user defined metrics and cluster level metrics in user alerts,
+higher availability can be achieved by hosting user defined alerting rules on the
+user workload Prometheus instances.
+
+This is enabled by setting a label on the PrometheusRule custom resource which is deployed by the user.
+Prometheus Operator then dispatches the deployment of this rule on the user workload monitoring Prometheus instance rather than on the Thanos Ruler instance.
+For this use case just one system is critical for alerts to work:
+
+1. User workload Prometheus
+
+### Console integration
+
+##### Developer perspective
+
+Alerts/Recording Rules Aggregation implies that the user potentially choses whether recording or alerting rules are being deployed on the user workload Prometheus or Thanos Ruler.
+In order to retrieve the total list user workload recording rules or alerting rules a fan-out request has to be executed against user workload Prometheus and Thanos Ruler
+and the response has to be merged for final display. The fan-out/merge proxy logic is suggested to be implemented in the console backend (being a Go based proxy server).
```

Comment:
I would personally welcome it if we could decouple this from the console backend so we could collaborate with the community on this even if people choose not to use OpenShift/OKD. I think this is ok for now, but I'd like us to extract this and regard it as technical debt. There is no reason in my opinion why all the features described here would not be possible in kube-prometheus.

Reply:
agreed, as discussed out of band, this could be a general purpose proxy.